lib/good_job/probe_server.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module GoodJob
+  class ProbeServer
+    RACK_SERVER = 'webrick'
+
+    def self.task_observer(time, output, thread_error) # rubocop:disable Lint/UnusedMethodArgument
+      return if thread_error.is_a? Concurrent::CancelledOperationError
+
+      GoodJob.on_thread_error.call(thread_error) if thread_error && GoodJob.on_thread_error.respond_to?(:call)
+    end
+
+    def initialize(port:)
+      @port = port
+    end
+
+    def start
+      @handler = Rack::Handler.get(RACK_SERVER)
```

Comment:
@bensheldon if you use `Rack::Handler.default` here you don't need the dependency on `webrick` and the handler can be set with an environment flag

Reply:
Ooh, that's interesting. When trying that change, it defaults to Puma, but Puma intercepts `ctrl-c`/`Interrupt` which becomes really problematic :-(

This isn't intended to be anything more than a container healthcheck, which makes me think that I should explore pasting in a "rack-compatible http server in 30 lines of ruby" snippet of code and avoid the dependency altogether.

Reply:
Ah I see, then webrick makes more sense 👍🏻